gmso/tests/test_mol2.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import pytest
+import unyt as u
+from unyt.testing import assert_allclose_units
+
+from gmso import Topology
+from gmso.formats.mol2 import from_mol2
+from gmso.tests.base_test import BaseTest
+from gmso.utils.io import get_fn
+
+
+class TestMol2(BaseTest):
+    def test_read_mol2(self):
+        top = Topology.load(get_fn("parmed.mol2"))
+        assert top.name == "parmed"
+        assert top.n_sites == 8
+        assert_allclose_units(
+            top.box.lengths,
+            ([8.2693, 7.9100, 6.6460] * u.Å).to("nm"),
+            rtol=1e-5,
+            atol=1e-8,
+        )
+        assert list(top.sites)[0].element.name == "carbon"
+        assert_allclose_units(
+            list(top.sites)[0].element.mass,
+            np.array(1.9944733e-26) * u.kg,
+            rtol=1e-5,
+            atol=1e-8,
+        )
+
+        top = Topology.load(get_fn("tip3p.mol2"))
+        assert top.name == "tip3p"
+        assert top.n_sites == 3
+        assert_allclose_units(
+            top.box.lengths, 3.0130 * np.ones(3) * u.Å, rtol=1e-5, atol=1e-8
+        )
+        positions_check = [
+            [0.061, 0.1, 0.1],
+            [0.017, 0.09, 0.177],
+            [0.011, 0.154, 0.04],
+        ]
+        for check, site in zip(positions_check, top.sites):
+            assert_allclose_units(
+                site.position,
+                check * u.nm,
+                rtol=1e-5,
+                atol=1e-8,
+            )
+
+        top = Topology.load(get_fn("vmd.mol2"))
+        assert top.name == "vmd"
+        assert top.n_sites == 6
+        assert len(top.bonds) == 5
+        assert top.bonds[0].connection_members[0] == top.sites[0]
+        assert top.box == None
+
+        match = "No charges were detected for site C with index 1"
+        with pytest.warns(UserWarning) as record:
```

Comment:
You dont need to manually match the warning or error messages. You can do that with the `match` keyword arguments
```suggestion
        with pytest.warns(UserWarning, match=r"No charges were detected for site C with index 1"):
```